src/webapp/src/auth/asyncAuthActions.tsx

```diff
@@ -0,0 +1,29 @@
+import { createAsyncThunk } from '@reduxjs/toolkit';
+import { AuthError } from '@firebase/auth-types';
+import { auth, provider } from '../firebase';
+import { User } from './types';
+
+export const signInUser = createAsyncThunk<
+  User,
+  void,
+  { rejectValue: AuthError }
+>('user/sigInUser', async (_, { rejectWithValue }) => {
+  try {
+    const { user } = await auth.signInWithPopup(provider);
+    const { displayName, uid, email } = user!;
```

Comment:
hi @paschalidi,
This is a more or less new feature from Typescript. It is called the "non-null assertion operator", we can use it as a post-fix to assert that the variable is not `null` or `undefined` in cases the type checker cannot conclude that.

 I used it here because the function `auth.signInWithPopup(provider)`  returns a value with a type `firebase.User | null` so I am telling the compiler I know user wont be `null` here.
 
 Did I explain it right? Or should I try again?

Reply:
It sounds about right. Thanks!